sdk/storage/storage-file-datalake/src/clients.ts

```diff
@@ -458,6 +460,15 @@ export class DataLakePathClient extends StorageClient {
     options.conditions = options.conditions || {};
     const { span, updatedOptions } = createSpan("DataLakePathClient-delete", options);
     try {
+      if (this.isTokenCredential === undefined) {
+        this.isTokenCredential = false;
+        this.pipeline.factories.forEach((factory) => {
```

Comment:
Will the check impact performance?
It might be better only go through pipeline to check the credential, when recursive is true.